CHANGELOG.md

```diff
@@ -0,0 +1,66 @@
+# Changelog
+
+## 7.0.0
+
+### A few classes have been renamed
+
+* `AuthConfig` -> `RegistryAuth`
+
+### Some methods have been renamed
+
+* `DockerClient.authConfig()` -> `DockerClient.registryAuth()`
+* `ContainerCreation.getWarnings()` -> `ContainerCreation.warnings()`
+* `ContainerConfig.getStopSignal()` -> `ContainerConfig.stopSignal()`
+* `ContainerConfig.volumes("/foo")` -> `ContainerConfig.addVolume("/foo")`
+* `ContainerConfig.volumes()` -> `ContainerConfig.volumeNames()`
+* `AttachedNetwork.globalIPv6Address()` -> `AttachedNetwork.globalIpv6Address()`
+* `TaskSpec.Builder`
+  * `withContainerSpec()` -> `containerSpec()`
+  * `withLogDriver()` -> `logDriver()`
+  * `withResources()` -> `resources()`
+  * `withRestartPolicy()` -> `restartPolicy()`
+  * `withPorts()` -> `addPort()`
+* `PortConfig.Builder`
+  * `withName()` -> `name()`
+  * `withProtocol()` -> `protocol()`
+  * `withPublishedPort()` -> `publishedPort()`
+  * `withTargetPort()` -> `targetPort()`
+* `RestartPolicy.Builder`
+  * `withCondition()` -> `condition()`
+  * `withDelay()` -> `delay()`
+  * `withMaxAttempts()` -> `maxAttempts()`
+* `ContainerSpec.Builder`
+  * `withImage()` -> `image()`
+  * `withCommands()` -> `command()`
+* `ServiceSpec.Builder`
+  * `withName()` -> `name()`
+  * `withTaskTemplate()` -> `taskTemplate()`
+  * `withServiceMode()` -> `serviceMode()`
+  * `withNetworks()` -> `networks()`
+  * `withEndpointSpec()` -> `endpointSpec()`
+  * `withUpdateConfig()` -> `updateConfig()`
+* `NetworkAttachmentConfig.Builder.withTarget()` -> `NetworkAttachmentConfig.Builder.target()`
+* `Service.Criteria.Builder`
+  * `withServiceId()` -> `serviceId()`
+  * `withServiceName()` -> `serviceName()`
+* `Task.Criteria.Builder`
+  * `withTaskId()` -> `taskId()`
+* `HostConfig.Bind`
+  * `new HostConfig.Bind.BuilderTo("to")` -> `HostConfig.Bind.BuilderTo.create("to")`
+  * `new HostConfig.Bind.BuilderFrom("from")` -> `HostConfig.Bind.BuilderFrom.create("from")`
+
+### Use public static factory methods instead of public constructors
+
+* `new RemovedImage()` -> `RemovedImage.create()`
+* `new ProgressMessage()` -> `ProgressMessage.builder()`
+
+### Misc
+
+```
+-    final Ipam ipam =
+-        Ipam.builder().driver("default").config("192.168.0.0/24", "192.168.0.0/24", "192.168.0.1")
+-            .build();
++    final IpamConfig ipamConfig =
++        IpamConfig.create("192.168.0.0/24", "192.168.0.0/24", "192.168.0.1");
++    final Ipam ipam = Ipam.create("default", singletonList(ipamConfig));
+```
```

Comment:
This is meant to be a diff documenting the breaking changes to `Ipam` and `IpamConfig`. I can add a sentence to clarify.